---
title: "Modelling complex traits with ancestral recombination graphs"
author: "Hanbin Lee"
affiliation: "University of Michigan, Ann Arbor"
email: "hblee@umich.edu"
venue: "Probabilistic Modelling in Genomics 2025"
date: "07 March 2025"
date-format: "MMM D, YYYY"
bibliography: ref.bib
format: 
  revealjs:
    slide-number: c/t
    width: 1600
    height: 900
    html-math-method: mathjax
    theme: white
    transition: slide
---

# Overview

::: {.incremental}
- We condition on the sample's ancestral recombination graph (ARG) 
- The full feature of the ARG is superfluous, only the local trees matter $\mathcal{T} = (\mathbf{V}, \mathbf{E})$
<p align="center">
	<img src="imgs/local_trees.svg" />
</p>	

- Drift and recombination are fixed, only mutation is left to vary
$$
\mathbf{y} \mid \mathcal{T} \; \sim \; ?
$$
:::

::: footer
Figure from [tskit docs](https://tskit.dev/tutorials/what_is.html)
:::


# Setup and derivation

The trait $\mathbf{y}$ is a linear function of the genotype $\mathbf{G}$
$$
\mathbf{y} = \mathbf{G}\boldsymbol{\beta} + \boldsymbol{\varepsilon}
$$
$\mathbf{y} \in \mathbb{R}^N$, $\mathbf{G} \in \mathbb{R}^{N \times P}$, 
$\boldsymbol{\beta} \in \mathbb{R}^P$, and $\boldsymbol{\varepsilon} \in \mathbb{R}^N$

::: {.incremental}
- $N \ll P$, it's impossible to fit a high-dimensional regression without additional assumptions
- Perfect collinearity of some columns in $\mathbf{G}$, or perfect linkage disequilibrium (LD)
- A population genetic solution: group variants into branches
:::

## Edge splitting

::: {.incremental}
- Nodes and edges are reused across multiple trees in a tree sequence
- Edges, in particular, may not have a unique set of samples along its span
- Salehi Nowbandegani and colleagues *bricked* the edges to divide them [@SalehiNowbandegani2023]
- Henceforth, we assume that edges are splitted to have a unique subtopology
$$
\mathbf{Z}_{ne} = \text{The number of haplotypes of individual $n$ that inherit $e$} 
$$
The overall matrix $\mathbf{Z}$ is an individual-edge design matrix.
:::


## Collapsing variants to edges

::: {.incremental}
- When does an individual posess a derived allele? (ancestral$=0$, derived$=1$)
- An individual is should be a descendant of an edge
- And that edge should have a mutation
- Let $\mathbf{1}_{ep}$ be the indicator *random* variable of a mutation at edge $e$ and position $p$
$$
\mathbf{G}_{np} = \sum_{e: p \in e} \mathbf{Z}_{np} \mathbf{1}_{ep} \quad \Leftrightarrow \quad \mathbf{G}_p = \sum_{e:p \in e} \mathbf{Z}_{e} \mathbf{1}_{ep}
$$
- Assumes that there are no parent-child mutation pairs, but allows *some* recurrent mutations
:::

## Exchange the summations


::: {.incremental}
- Recall $\mathbf{G}_p = \sum_{e:p \in e} \mathbf{Z}_{e} \mathbf{1}_{ep}$
$$
\begin{aligned}
\mathbf{y} &= \sum_{p=1}^P \mathbf{G}_p \boldsymbol{\beta}_p + \boldsymbol{\varepsilon} 
		= \textcolor{grey}{\sum_{p=1}^P \sum_{e:p \in e}} \mathbf{Z}_e \boldsymbol{\beta}_p \mathbf{1}_{ep} + \boldsymbol{\varepsilon} \\
		&= \textcolor{red}{\sum_{e=1}^E \sum_{p:p \in e}} \mathbf{Z}_e \boldsymbol{\beta}_p \mathbf{1}_{ep} + \boldsymbol{\varepsilon} 
		= \sum_{e=1}^E \mathbf{Z}_e \textcolor{blue}{\left(\sum_{p:p\in e} \boldsymbol{\beta}_p \mathbf{1}_{ep} \right)} + \boldsymbol{\varepsilon} \\
		&= \sum_{e=1}^E \mathbf{Z}_e \textcolor{blue}{\boldsymbol{\upsilon}_e} + \boldsymbol{\varepsilon}
		= \mathbf{Z} \boldsymbol{\upsilon} + \boldsymbol{\varepsilon}
\end{aligned}
$$

- $\boldsymbol{\upsilon}$ is a random variable made up of mutation-driven random variables $\mathbf{1}_{ep}$!
:::

# Ancestral recombination graph linear mixed model (ARG-LMM)

Split $\boldsymbol{\upsilon}$ to $\mathbf{u} = \boldsymbol{\upsilon} - \mathrm{E}\boldsymbol{\upsilon}$ and $\mathbf{f}= \mathrm{E}\boldsymbol{\upsilon}$

$$
\mathbf{y} = \mathbf{Z} \mathbf{u} + \mathbf{Z} \mathbf{f} + \boldsymbol{\varepsilon}
$$

Let's characterize $\mathbf{u}$ (the random effects) and $\mathbf{f}$ (the fixed effects)

## Random effects are independent

::: {.incremental}
- Independent entries of random effects a key assumption of linear mixed models
- This can be *proved* in ARG-LMM, instead of assuming it
$$
	\mathrm{Cov}( \mathbf{u}_e, \mathbf{u}_{e'} ) = \sum_{p \in e,e'} \boldsymbol{\beta}_p^2 \mathrm{Cov}(\mathbf{1}_{ep}, \mathbf{1}_{e'p})
$$

- The covariance between the indicators are higher-order terms of mutation rates, so we ignore it [@Wakeley2008]
$$
	\begin{aligned}
	\mathrm{Cov}(\mathbf{1}_{ep}, \mathbf{1}_{e'p}) &= \mathrm{E}[\mathbf{1}_{ep}\mathbf{1}_{e'p}] - \mathrm{E}[\mathbf{1}_{e'p}]\mathrm{E}[\mathbf{1}_{ep}] \\
		&= 0 -l_eu_{ep}l_{e'}u_{e'p} \approx 0
	\end{aligned}
$$
where $l_e$ is the (time-)length of edge $e$.

:::


## The marginal distribution of $\mathbf{u}_e$?

::: {.incremental}
- The Gaussian prior on random effects is a popular choice
- One might be tempted to invoke the central limit theorem to $\mathbf{u}_e$ (sum of indicators)
	$$
		\mathbf{u}_e \bigg/ \sqrt{l_e s_e} \cdot \sqrt{ \frac{1}{s_e} \sum_{p:p \in e} \boldsymbol{\beta}_p^2 u_{ep} } 
		\rightarrow N(0,1^2) \text{ as } s_e \rightarrow \infty
	$$
	where $s_e$ is the span (in base pairs) of edge $e$

- The convergence is unlikely to be fast enough given the small value of $\mathbf{E}[\mathbf{1}_{ep}]$ (Berry-Esseen).

- Fortunately, the variance is computable and is 
	$$
		\mathrm{Var}(\mathbf{u}_e) = l_e s_e \cdot \frac{1}{s_e} \sum_{p:p \in e} \boldsymbol{\beta}_p^2 u_{ep}
	$$
:::

## More on $\mathrm{Var}(\mathbf{u}_e)$

::: {.incremental}
- The weight $\mathrm{Var}(\mathbf{u}_e)$ has two components
- The area $l_e s_e$ 
- Mutation rate-weighted squared average of effect sizes
	$$
		\tau_e^2 = \frac{1}{s_e} \sum_{p:p \in e} \boldsymbol{\beta}_p^2 u_{ep}
	$$
- As a measure of functional significance, variance components confounded by the area
:::

## Fixed effects are constant under neutrality

::: {.incremental}

- Suppose that $u_{ep}=u_p$, i.e., the mutation rate is constant across edges for a given position
$$
\begin{aligned}
	\left[ \mathbf{Z} \mathbf{f} \right]_n &= \sum_{e=1}^E \mathbf{Z}_{ne} \mathbf{E}
	\left[ \sum_{p: p \in e} \boldsymbol{\beta}_p\mathbf{1}_{ep} \right] \\
		&= \sum_{p=1}^P \boldsymbol{\beta}_p u_p 
		\left(
				\sum_{e: p \in e} \mathbf{Z}_{ne} l_e
		\right) \\
	&= \sum_{p=1}^P \boldsymbol{\beta}u_p \cdot 2t_{\mathrm{root},p} = \text{const. resp. to $n$}
\end{aligned}
$$

- An intercept is enough to account for the fixed effects $\mathbf{Z}\mathbf{f}$ under this condition

- The assumption is standard in neutral settings
$$
	\text{\textcolor{red}{Conjecture:} selection $\Rightarrow$ fixed effects?}
$$

:::

# $\textsf{tslmm}$, fitting ARG-LMM to tree sequences

::: {.incremental}
- $\textsf{tslmm}$ utilizes an efficient genetic relatedness matrix - vector product to fit the restricted maximum likelihood (REML) objective
- Assumes
$$
	\frac{1}{s_e} \sum_{p: p \in e} \boldsymbol{\beta}^2 u_{ep} = \tau_e^2 = \tau^2
$$
for all edges $e \in \mathbf{E}$.
- Approximately true provided the edge is long enough


:::

## Runtime

<p align="center">
<img src="imgs/runtime.svg" />
</p>

## Precision
<p align="center">
<img src="imgs/estimation_true.svg" />
</p>
<p align="center">
<img src="imgs/estimation_infer.svg" />
</p>

## Best linear unbiased prediction (BLUP)

<p align="center">
<img src="imgs/prediction.svg" />
</p>

## References

::: {#refs}
:::


