---
title: "Modelling complex traits with ancestral recombination graphs"
author: "Hanbin Lee"
affiliation: "University of Michigan, Ann Arbor"
email: "hblee@umich.edu"
venue: "Probabilistic Modelling in Genomics 2025"
date: "07 March 2025"
date-format: "MMM D, YYYY"
bibliography: ref.bib
format: 
  revealjs:
    slide-number: c/t
    width: 1600
    height: 900
    html-math-method: mathjax
    theme: white
    transition: slide
---

# Overview

::: {.incremental}
- We condition on the sample's ancestral recombination graph (ARG) 
- The full feature of the ARG is superfluous, only the local trees matter $\mathcal{T} = (\mathbf{V}, \mathbf{E})$
![](imgs/local_trees.svg)

- Drift and recombination are fixed, only mutation is left to vary
$$
\mathbf{y} \mid \mathcal{T} \; \sim \; ?
$$
:::

::: footer
Figure from [tskit docs](https://tskit.dev/tutorials/what_is.html)
:::


# Setup and derivation

The trait is a linear function of $y$
$$
\mathbf{y} = \mathbf{G}\boldsymbol{\beta} + \boldsymbol{\varepsilon}
$$
$\mathbf{y} \in \mathbb{R}^N$, $\mathbf{G} \in \mathbb{R}^{N \times P}$, 
$\boldsymbol{\beta} \in \mathbb{R}^P$, and $\boldsymbol{\varepsilon} \in \mathbb{R}^N$

::: {.incremental}
- $N \ll P$, high-dimensional regression needs a fix
- Perfect collinearity of some columns in $\mathbf{G}$, or perfect LD
- A population genetic solution: group variants into branches
:::

## Edge splitting

::: {.incremental}
- Nodes and edges are reused across multiple trees in a tree sequence
- Edges, in particular, may not have a unique set of samples along its span
- Salehi Nowbandegani and colleagues *bricked* the edges to divide them [@SalehiNowbandegani2023]
- Henceforth, we assume that edges are splitted to have a unique subtopology
$$
\mathbf{Z}_{ne} = \text{The number of haplotypes of individual $n$ that inherit $e$} 
$$
The overall matrix $\mathbf{Z}$ is an individual-edge design matrix.
:::


## How to gather edges

::: {.incremental}
- When does an individual posess a derived allele? (ancestral=$0$, derived=$1$)
- An individual is should be a descendant of an edge
- And that edge should have a mutation
- Let $\mathbf{1}_{ep}$ be the indicator *random* variable of a mutation at edge $e$ and position $p$
$$
\mathbf{G}_{np} = \sum_{e: p \in e} \mathbf{Z}_{np} \mathbf{1}_{ep} \quad \Leftrightarrow \quad \mathbf{G}_p = \sum_{e:p \in e} \mathbf{Z}_{e} \mathbf{1}_{ep}
$$
- Assumes that there is no parent-child mutation pairs, but allows recurrent mutations
:::

## Exchange the summations


::: {.incremental}
- Recall $\mathbf{G}_p = \sum_{e:p \in e} \mathbf{Z}_{e} \mathbf{1}_{ep}$
$$
\begin{aligned}
\mathbf{y} &= \sum_{p=1}^P \mathbf{G}_p \boldsymbol{\beta}_p + \boldsymbol{\varepsilon} 
		= \textcolor{grey}{\sum_{p=1}^P \sum_{e:p \in e}} \mathbf{Z}_e \boldsymbol{\beta}_p \mathbf{1}_{ep} + \boldsymbol{\varepsilon} \\
		&= \textcolor{red}{\sum_{e=1}^E \sum_{p:p \in e}} \mathbf{Z}_e \boldsymbol{\beta}_p \mathbf{1}_{ep} + \boldsymbol{\varepsilon} 
		= \sum_{e=1}^E \mathbf{Z}_e \textcolor{blue}{\left(\sum_{p:p\in e} \boldsymbol{\beta}_p \mathbf{1}_{ep} \right)} + \boldsymbol{\varepsilon} \\
		&= \sum_{e=1}^E \mathbf{Z}_e \textcolor{blue}{\boldsymbol{\upsilon}_e} + \boldsymbol{\varepsilon}
		= \mathbf{Z} \boldsymbol{\upsilon} + \boldsymbol{\varepsilon}
\end{aligned}
$$

- $\boldsymbol{\upsilon}$ is a random variable made up of mutation-driven random variables $\mathbf{1}_ep$!
:::

# Ancestral recombination graph linear mixed model (ARG-LMM)

Split $\boldsymbol{\upsiilon}$ to $\mathbf{u} = \boldsymbol{\upsilon} - \mathrm{E}\boldsymbol{\upsilon}$ and $\mathbf{f}= \mathrm{E}\boldsymbol{\upsilon}$

$$
\mathbf{y} = \mathbf{Z} \mathbf{u} + \mathbf{Z} \mathbf{f} + \boldsymbol{\varepsilon}
$$

Let's characterize $\mathbf{u}$ (the random effects) and $\mathbf{f}$ (the fixed effects)

## Random effects are independent


## The marginal distribution of $\mathbf{u}_e$?


## Fixed effects are constant under neutrality



## References

::: {#refs}
:::


