---
layout: page
title:  "Querying RSID from Chromosome:Position (2)"
subtitle: ""
date:	2023-02-20
comments: true
categories: ["Bioinformatics"]
---

In the last [post](https://hanbin973.github.io/bioinformatics/2023/02/20/rsid.html), I explained how to add RSIDs to GWAS summary statistics.
Although the final RSID assignment step was fast enough for processing large data, the overall process was quite cumbersome for several reasons.
Most of all, it first extracted the `chr:pos` to a separate file and then assigned the RSIDs.
Therefore, the user had to combine the original sumstats file with the `chr:pos`-RSID file manually after obtaining the RSIDs for the variants in `chr:pos`.

This post explains how to avoid these steps and write a single command that does everything you need.
The magic comes from using pipes (`|`) that combine multiple commands in a shell.
This is also helpful for debugging your script in the shell by throwing the `head` command between the lines.
Additionally, I explain how to move between different genome builds (e.g. between `hg19` and `hg38`) using the `liftOver` command.

The required programs and the files is listed below
1. [`liftOver` command-line tool](http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/). Find the `liftOver` binary in the link. The link contains `x86_64-linux` binaries so look at other directories for other operating systems.
2. [`bedops` command-line tool](https://bedops.readthedocs.io/en/latest/index.html). `bedops` is a collection of many command-line tools. The one we are gonna use is `bedmap`.
3. `liftOver` chain files for [`hg38`](https://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/) and [`hg19`](https://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/). Files for other genome builds can be found in the parent directories of the link. Note that the genome build here referes to the genome build of the input file. Some basic examples for using these chain files can be found [here](https://genome.sph.umich.edu/wiki/LiftOver).
4. dbSNP reference for [`hg38`](https://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/) and [`hg19`](http://hgdownload.soe.ucsc.edu/goldenPath/hg19/database/). The file we use today is `snp144.txt.gz` which is a full list of SNPs in the dbSNP version 144. The list contains the SNPs together with their annotations such as RSID, `chr:pos`, variant type and many more.

After installing the programs (simply put the binary files in to your `~/bin` folder) and saving the files, my folder currently looks like
```
├── chains
│   ├── hg19ToHg38.over.chain.gz -> # downloaded from `hg19` chains folder
│   └── hg38ToHg19.over.chain.gz # downloaded from `hg38` chains folder
└── dbSNP
    └── snp144.txt.gz # downloaded from `hg19` dbSNP
```


## Step 1: Preparing the files.

Both `liftOver` and `bedops` operate on a `bed` file. 
In the [UCSC wiki](https://genome.ucsc.edu/FAQ/FAQformat.html), the precise description of the format is given.
12 columns are explained but only the first three is actually important in our application.
Furthermore, the programs we use today don't even care what appears after the first three. 

First, we convert `snp144.txt.gz` into a proper `bed` file.
Before the conversion, let's see how the file looks like.
The file is compressed so a naive way to see the content is to unzip it first.
However, as you'll notice, the file is huge so unzipping the whole file isn't a nice choice.

The nice part of working in a shell is that many programs operates in a line-by-line manner.
This means that the program processes the file line-by-line and throws out the output.
Only the first few lines will tell you how the file looks like so we could unzip the first few lines.
This can be done using the combination of `zcat` and `head` commands.
```
zcat snp144.txt.gz | head -n 10 

585     chr1    10019   10020   rs775809821     0       +       A       A       -/A     genomic deletion        unknownnear-gene-5      exact   1               1       SSMP,   0
585     chr1    10055   10055   rs768019142     0       +       -       -       -/A     genomic insertion       unknownnear-gene-5      between 1               1       SSMP,   0
585     chr1    10107   10108   rs62651026      0       +       C       C       C/T     genomic single  unknown 0      near-gene-5      exact   1               1       BCMHGSC_JDW,    0
585     chr1    10108   10109   rs376007522     0       +       A       A       A/T     genomic single  unknown 0      near-gene-5      exact   1               1       BILGI_BIOE,     0
585     chr1    10138   10139   rs368469931     0       +       A       A       A/T     genomic single  unknown 0      near-gene-5      exact   1               1       BILGI_BIOE,     0
```
There are total 20 columns.
    
        





