---
layout: page
title:  "Querying RSID from Chromosome:Position (2)"
subtitle: ""
date:	2023-02-20
comments: true
categories: ["Bioinformatics"]
---

In the last [post](https://hanbin973.github.io/bioinformatics/2023/02/20/rsid.html), I explained how to add RSIDs to GWAS summary statistics.
Although the final RSID assignment step was fast enough for processing large data, the overall process was quite cumbersome for several reasons.
Most of all, it first extracted the `chr:pos` to a separate file and then assigned the RSIDs.
Therefore, the user had to combine the original sumstats file with the `chr:pos`-RSID file manually after obtaining the RSIDs for the variants in `chr:pos`.

This post explains how to avoid these steps and write a single command that does everything you need.
The magic comes from using piples (`|`) that combines multiple commands in a shell.
This is also helpful for debugging your script in the shell by throwing the `head` command between the lines.
Additionally, I explain how to move between different genome builds (e.g. between `hg19` and `hg38`) using the `liftOver` command.

The required programs and the files is listed below
1. [`liftOver` command-line tool](http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/). Find the `liftOver` binary in the link. The link contains `x86_64-linux` binaries so look at other directories for other operating systems.
2. [`bedops` command-line tool](https://bedops.readthedocs.io/en/latest/index.html). `bedops` is a collection of many command-line tools. The one we are gonna use is `bedmap`.
3. `liftOver` chain files for [`hg38`](https://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/) and [`hg19`](https://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/). Files for other genome builds can be found in the parent directories of the link. Note that the genome build here referes to the genome build of the input file. Some basic examples for using these chain files can be found [here](https://genome.sph.umich.edu/wiki/LiftOver).
4. dbSNP reference for [`hg38`](https://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/) and [`hg19`](http://hgdownload.soe.ucsc.edu/goldenPath/hg19/database/). The file we use today is `snp144.txt.gz` which is a full list of SNPs in the dbSNP version 144. The list contains the SNPs together with their annotations such as RSID, `chr:pos`, variant type and many more.





